flink/v1.17/flink/src/main/java/org/apache/iceberg/flink/sink/shuffle/AggregateDataStatistics.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iceberg.flink.sink.shuffle;
+
+import java.io.Serializable;
+import java.util.Set;
+import org.apache.iceberg.relocated.com.google.common.base.MoreObjects;
+import org.apache.iceberg.relocated.com.google.common.base.Preconditions;
+import org.apache.iceberg.relocated.com.google.common.collect.Sets;
+
+class AggregateDataStatistics<K> implements Serializable {
```

Comment:
nit: maybe a different name. `AggregateDataStatistics` sounds like it should implement `DataStatistics` interface, which it doesn't. Maybe `ReportedStatisticsTracker`? also add a simple Javadoc to briefly explain what this class does. E.g., explain it only tracks the aggregated stats and use a Set<Integer> to track which subtask has reported.

Reply:
> also add a simple Javadoc to briefly explain what this class does. E.g., explain it only tracks the aggregated stats and use a Set to track which subtask has reported.

+1.
If possible, we should add doc to each new class to help people quickly understand what it does and how to use it properly.

Reply:
> maybe a different name. `AggregateDataStatistics` sounds like it should implement `DataStatistics` interface, which it doesn't. Maybe `ReportedStatisticsTracker`?

Maybe `GlobalStatisticsAccumulator `?